apps/src/lib/tools/jsdebugger/DebugConsole.jsx

```diff
@@ -211,11 +242,33 @@ export default connect(
 
     displayOutputToConsole() {
       if (this.props.logOutput.size > 0) {
-        return this.props.logOutput.map((output, i) => {
-          if (typeof output === 'object') {
-            output = output.toJS();
+        return this.props.logOutput.map((rowValue, i) => {
+          try {
+            rowValue = rowValue.toJS();
+          } catch (error) {
+            return <Inspector key={i} data={rowValue} />;
+          }
+          if (rowValue.input) {
+            return <div key={i}>&gt; {rowValue.input}</div>;
+          } else if (
+            rowValue.output ||
+            FALSY_VALUES.has(rowValue.output) ||
+            (rowValue.output === undefined &&
+              (rowValue.fromConsoleLog || rowValue.undefinedInput))
```

Comment:
Given how complicated this conditional statement is, I'd recommend moving it into its own function. i.e. shouldPrintOutput. Then you can spend a bit more space making it more readable.

Reply:
Yes, that makes sense! I'll work on it.